pkg/build/api/v1/types.go

```diff
@@ -176,8 +176,89 @@ type BuildStatus struct {
 
 	// output describes the Docker image the build has produced.
 	Output BuildStatusOutput `json:"output,omitempty" protobuf:"bytes,10,opt,name=output"`
+
+	// steps contains details about each build step.
+	Stages Stages `json:"stages,omitempty" protobuf:"bytes,11,opt,name=stages"`
+}
+
+// Stages is a slice of build stages that have been recorded.
+type Stages []StageInfo
```

Comment:
Generally we don't do this unless there's a reason that has to do with protobuf serialization (null vs empty behavior), and that doesn't apply here.  Recommend removing this alias type.